tests/suites/test_suite_md.function

```diff
@@ -49,55 +49,53 @@ void md_null_args()
 
     mbedtls_md_init(&ctx);
 
-    TEST_ASSERT(mbedtls_md_get_size(NULL) == 0);
+    TEST_EQUAL(0, mbedtls_md_get_size(NULL));
```

Comment:
Why not change all occurrences of `TEST_ASSERT` to `TEST_EQUAL`?

Reply:
Because I think `TEST_EQUAL` mostly makes sense for numerical data, as it's converting each side to `unsigned long long` before printing both values in case of failures.

Reply:
Indeed `TEST_EQUAL` is only for integral types (of any size, as a practical matter, although the current implementation only supports sizes up to 64 bits). There's no similar macro for pointers because we very rarely compare pointers, and having a printout of the pointer value on failure would not be very useful.

However it would make sense to change things like the line below to `TEST_EQUAL(mbedtls_md_get_type(NULL), MBEDTLS_MD_NONE);`.

Reply:
I was suspecting that `TEST_EQUAL` is not designed for pointers, but we still have numeric checks for:
- `mbedtls_md_get_type`
- `mbedtls_test_hexcmp`.

Reply:
`mbedtls_test_hexcmp` shouldn't exist at all. It's legacy from the pre-2018 test framework where test functions received hex data and converted it to binary, and comparing actual output with expected output was often done by converting the actual output to hex. Now test functions normally receive output as binary, and should use `ASSERT_COMPARE` to compare the actual output with expected output. We have a [technical debt task](https://github.com/Mbed-TLS/mbedtls/issues/3685) to get rid of `mbedtls_test_hexcmp`.

Since `mbedtls_test_hexcmp` just returns a boolean-like result, it isn't useful to use `TEST_EQUAL` on it. `TEST_EQUAL` is useful when there's more than one possible value on failure, but when `TEST_ASSERT(mbedtls_test_hexcmp(x, y) == 0` fails, you already know that it means that `x` is not equal to `y` and `TEST_EQUAL` won't tell you more (whereas `ASSERT_COMPARE` might).

Reply:
Ah indeed I missed the `get_type()` tests, thank you Przemek for noticing! I'll fix those.

For `test_hexcmp()`, it's as Gilles said: nothing to be gained by printing the values. Also, I considered creating `TEST_MEMCMP()` that would print the contents of the buffers on failure, but decided against it (for now) as there would be questions what to do if the output's too large for the fixed-size buffers we have for `line1` and `line2` in the test results structure, and I decided this was taking us too far for this PR. That's the other reason why `test_hexcmp()` was done at the same time as the others.

I've also considered changing to `TEST_ASSERT(0 == memcmp(...));` to make things clearer (we all know what `memcmp()` is, unlike `mbedtls_test_hexcmp()` that I had to look up), but didn't do it because I wasn't sure it would be useful. Wdyt?

Could also create `TEST_MEMCMP()` right now, but for now not make it print anything special when the buffers are different - that part could be added later.

Reply:
`TEST_MEMCMP` already exists (and doesn't print the content differences, but ensures that the sizes are compared — that was my first priority when I introduced it, after tests that called `memcmp` but forgot to compare sizes), but it's called `ASSERT_COMPARE`, which is a [separate issue](https://github.com/Mbed-TLS/mbedtls/pull/6253).

Reply:
Oh, so now I'm really glad I didn't create `TEST_MEMCMP()` :) I've changed to use `ASSERT_COMPARE()` and fixed the place I missed with `get_type()` (actually I chnaged most of them except one, so that was clearly a mistake). The remaining uses of `TEST_ASSERT()` are now only pointer comparisons.